pkg/package-server/server/server.go

```diff
@@ -154,52 +154,59 @@ func (o *PackageServerOptions) Run(stopCh <-chan struct{}) error {
 		return fmt.Errorf("unable to construct lister client config: %v", err)
 	}
 
-	// set up the informers
 	kubeClient, err := kubernetes.NewForConfig(clientConfig)
 	if err != nil {
 		return fmt.Errorf("unable to construct lister client: %v", err)
 	}
 
-	// create a new client for OLM types (CRs)
 	crClient, err := client.NewClient(o.Kubeconfig)
 	if err != nil {
 		return err
 	}
 
 	log.Infof("package-server configured to watch namespaces %v", o.WatchedNamespaces)
 
-	// Create an informer for each catalog namespace
 	catsrcSharedIndexInformers := []cache.SharedIndexInformer{}
 	for _, namespace := range o.WatchedNamespaces {
 		nsInformerFactory := externalversions.NewSharedInformerFactoryWithOptions(crClient, o.WakeupInterval, externalversions.WithNamespace(namespace))
 		catsrcSharedIndexInformers = append(catsrcSharedIndexInformers, nsInformerFactory.Operators().V1alpha1().CatalogSources().Informer())
 	}
 
-	// Create a new queueinformer-based operator.
 	queueOperator, err := queueinformer.NewOperator(o.Kubeconfig)
 	if err != nil {
 		return err
 	}
 
 	sourceProvider := provider.NewInMemoryProvider(catsrcSharedIndexInformers, queueOperator)
 
-	// inject the providers into the config
-	config.ProviderConfig.Provider = sourceProvider
+	go func() {
```

Comment:
Nice! My only reservation is that without this goroutine, the provider will be blocked until someone calls watch. I almost feel like this belongs in a `Provider.Run(stopCh <-chan struct{})`, just to keep it coupled with the provider.

Reply:
Another thought; we could also have the provider store three separate channel slices, `add, modify, delete []chan *PackageManifest`. Each new call to subscribe could make and append the appropriate channels to these slices. When the provider detects an event, it should be emitted to all channels in the appropriate slice. 

Now that I'm thinking about this more deeply. It looks like having a single channel for each event type will cause issues of its own. Channels deliver _once_, so if mutiple goroutines are reading from the channels returned from subscribe, only one of them will get a sent event.